src/backend/distributed/executor/local_executor.c

```diff
@@ -235,6 +237,147 @@ ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterT
 }
 
 
+/*
+ * ExecuteLocalUtilityTaskList executes a list of tasks locally. This function
+ * also logs local execution notice for each task and sets
+ * TransactionAccessedLocalPlacement to true for next set of possible queries
+ * & commands within the current transaction block. See the comment in function.
+ */
+void
+ExecuteLocalUtilityTaskList(List *localTaskList)
+{
+	Task *localTask = NULL;
+
+	foreach_ptr(localTask, localTaskList)
+	{
+		const char *localTaskQueryCommand = TaskQueryString(localTask);
+
+		if (localTask->anchorShardId != INVALID_SHARD_ID)
```

Comment:
when would this check return false?

Reply:
preparing some tasks, we just set the anchor shard id to be `INVALID_SHARD_ID`, without knowing the exact reason behind it

Reply:
no such tasks for utility commands, converting it to an assertion then